reactor-netty-core/src/main/java/reactor/netty/channel/ChannelOperations.java

```diff
@@ -96,9 +96,11 @@ public static void addMetricsHandler(Channel ch, ChannelMetricsRecorder recorder
 		if (remote == null) {
 			remote = ch.remoteAddress();
 		}
-		ChannelHandler handler = recorder instanceof ContextAwareChannelMetricsRecorder ?
-				new ContextAwareChannelMetricsHandler((ContextAwareChannelMetricsRecorder) recorder, remote, onServer) :
-				new ChannelMetricsHandler(recorder, remote, onServer);
+		ChannelHandler handler = recorder instanceof MicrometerChannelMetricsRecorder ?
```

Comment:
> one small nitpick, otherwise looks good.
> note that this is targetting `documented-meters` branch, I assume this is on purpose?

Yes I'll re-target before merging